### PR TITLE
Refactor removeDirectory; Check errors of operations

### PR DESCRIPTION
**Pull request by bscheshirwork**
So... This is can be a #62 but this is separated now... :smile_cat:

### CODE REVIEW

**Comment by bscheshirwork** [comment]
Work In Progress. Please do not merge now ^^

File: GalleryBehavior.php [hunk 1]
```diff
@@ -252,31 +268,48 @@ public function replaceImage($imageId, $path)
...
+     * @return bool
+     */
     private function createFolders($filePath)
     {
-        return FileHelper::createDirectory(FileHelper::normalizePath(dirname($filePath)), 0777);
+        try {
+            return FileHelper::createDirectory(FileHelper::normalizePath(dirname($filePath)), 0777);
+        } catch (\yii\base\Exception $exception) {
+            return false;
+        }
```
Comment (zxbodya) [inline GalleryBehavior.php]: same as above

File: GalleryBehavior.php [hunk 2]
```diff
@@ -252,31 +268,48 @@ public function replaceImage($imageId, $path)
...
-        $pk = $this->owner->getPrimaryKey();
-        if (is_array($pk)) {
-            return implode($this->pkGlue, $pk);
-        } else {
-            return $pk;
+        try {
+            FileHelper::removeDirectory(dirname($filePath));
+        } catch (\yii\base\ErrorException $exception) {
+            return false;
         }
```
Comment (zxbodya) [inline GalleryBehavior.php]: same as above, might be better to not catch this exception

File: GalleryBehavior.php [hunk 3]
```diff
@@ -252,31 +268,48 @@ public function replaceImage($imageId, $path)
...
+     * @return bool
+     */
     private function removeFile($fileName)
     {
-        return FileHelper::unlink($fileName);
+        try {
+            return FileHelper::unlink($fileName);
+        } catch (\yii\base\ErrorException $exception) {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (bscheshirwork) [reply GalleryBehavior.php]: Catch today: first erase directory from filesystem; second delete from web

File: GalleryBehavior.php (hunk 2 above)
Comment (bscheshirwork) [reply GalleryBehavior.php]: Same case: first rm dir, second try to delete this dir again

File: GalleryBehavior.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (zxbodya) [reply GalleryBehavior.php]: not sure I got it, it about case when two people trying concurrently to remove save thing?

File: GalleryBehavior.php (hunk 1 above)
Comment (bscheshirwork) [reply GalleryBehavior.php]: For this case cath some errors of chmod. Also we can improve it later

File: GalleryBehavior.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (bscheshirwork) [reply GalleryBehavior.php]: Нет, достаточно просто каким-то образом удалить файлы до удаления записей из базы

[the remark this answers is not in the document]
Comment (bscheshirwork) [reply GalleryBehavior.php]: Достатньо видалити з системи файл. Будь яким чином.

[the remark this answers is not in the document]
Comment (bscheshirwork) [reply GalleryBehavior.php]: Anyway this is same to suppress errors throw @

**Pull request closed without merging**